### ISSUE TITLE
Ubuntu 20.04 Codedeploy v1.1.2

### ISSUE THREAD

**Issue by RichardCorcoran** [aws/aws-codedeploy-agent#264]
Codedeploy-agent v1.1.2 claims to support Ubuntu 20.04, however when trying to install it on a fresh install of Ubuntu 20.04 the same error as with previous versions occurs.
This error is:
$ sudo ./install
Traceback (most recent call last):
        8: from ./install:34:in `<main>'
        7: from ./install:34:in `new'
        6: from /usr/lib/ruby/2.7.0/logger.rb:389:in `initialize'
        5: from /usr/lib/ruby/2.7.0/logger.rb:389:in `new'
        4: from /usr/lib/ruby/2.7.0/logger/log_device.rb:18:in `initialize'
        3: from /usr/lib/ruby/2.7.0/logger/log_device.rb:83:in `set_dev'
        2: from ./install:22:in `method_missing'
        1: from ./install:22:in `map'
./install:23:in `block in method_missing': undefined method `path' for #<IO:<STDOUT>> (NoMethodError)

**Comment by brblck** [comment]
Thank you for the report @RichardCorcoran. Our own internal integration tests for Ubuntu 20.04 didn't run into this issue, but we've seen a few similar reports and we'd like to investigate more.

Can you share with us:
- Relevant information about your Ruby 2.7 installation? (eg. Is it the system ruby and contents of `ruby -v`)
- What steps can we take to reproduce the issue?

**Comment by luiszimmermann** [comment]
This happens with me in Ubuntu 20.04, 

* Launch instance with the `Ubuntu Server 20.04 LTS (HVM), SSD Volume Type - ami-083aa2af86ff2bd11` (sa-east-1)
* `apt update & apt full-upgrade`
* `apt install ruby`
  * ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]
* `wget https://aws-codedeploy-sa-east-1.s3.amazonaws.com/latest/install`
* chmod +x
* `./install auto` output:
```
Traceback (most recent call last):
	8: from ./install:34:in `<main>'
	7: from ./install:34:in `new'
	6: from /usr/lib/ruby/2.7.0/logger.rb:389:in `initialize'
	5: from /usr/lib/ruby/2.7.0/logger.rb:389:in `new'
	4: from /usr/lib/ruby/2.7.0/logger/log_device.rb:18:in `initialize'
	3: from /usr/lib/ruby/2.7.0/logger/log_device.rb:83:in `set_dev'
	2: from ./install:22:in `method_missing'
	1: from ./install:22:in `map'
./install:23:in `block in method_missing': undefined method `path' for #<IO:<STDOUT>> (NoMethodError)
```

**Comment by luiszimmermann** [comment]
Tried with ruby 2.7.1 (installed with rbenv) `ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]`, same error.

**Comment by luiszimmermann** [comment]
Removing the usage of `Proxy` in line 34 made the script to work, but the output only goes to the tmp file and not in the terminal. At least the installation works.

```ruby
@log = Logger.new(File.open(log_file_path, 'a+'))
```

**Comment by paulthorburn** [comment]
I had submitted and a bug with the same issue, my fault for not providing more detail in it but i get the same error as Luis on Ubuntu 20.04.  I used https://docs.aws.amazon.com/codedeploy/latest/userguide/codedeploy-agent-operations-install-ubuntu.html as a reference but it has not been updated for 20.04 yet. I was able to install CodeDepoly with dpkg which seems to work.

**Broken:**
sudo apt-get install ruby
```
vagrant@ubuntu-focal:~$ ruby --version
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]
```
wget https://aws-codedeploy-us-west-2.s3.us-west-2.amazonaws.com/latest/install
chmod +x ./install
sudo ./install auto
```
Traceback (most recent call last):
	8: from ./install:34:in `<main>'
	7: from ./install:34:in `new'
	6: from /usr/lib/ruby/2.7.0/logger.rb:389:in `initialize'
	5: from /usr/lib/ruby/2.7.0/logger.rb:389:in `new'
	4: from /usr/lib/ruby/2.7.0/logger/log_device.rb:18:in `initialize'
	3: from /usr/lib/ruby/2.7.0/logger/log_device.rb:83:in `set_dev'
	2: from ./install:22:in `method_missing'
	1: from ./install:22:in `map'
./install:23:in `block in method_missing': undefined method `path' for #<IO:<STDOUT>> (NoMethodError)

```

**Working:**
apt-get update
apt-get install -y ruby
wget https://aws-codedeploy-us-east-1.s3.amazonaws.com/releases/codedeploy-agent_1.2.0-1866_all.deb
dpkg -i codedeploy-agent_1.2.0-1866_all.deb
systemctl start codedeploy-agent
systemctl enable codedeploy-agent

**Comment by philstrong** [comment]
Working on a fix, should be part of next release of the agent

**Comment by philstrong** [comment]
Issue is fixed in 1.3.2
